vm/devices/storage/disk_layered/src/lib.rs

```diff
@@ -455,7 +458,7 @@ impl<T: LayerAttach> DynLayerAttach for T {
 /// LayerAttach for T` which simply returns `Self` during the state transition.
 pub trait LayerAttach: 'static + Send + Sync + Inspect {
     /// Error returned if on attach failure.
-    type Error: std::error::Error + Send + Sync + 'static;
+    type Error: Into<anyhow::Error>;
```

Comment:
Use `Into<Box<dyn Error + Send + Sync>>`. This is sufficient for `Into<anyhow::Error>` and it avoids leaking `anyhow` into the public interface.

Reply:
I was trying to do that, but for whatever reason, I couldn't for the life of me get anyhow::Error to type-check against the `Into<Box<dyn Error + Send + Sync>>` bounds. I def saw that we did something like that elsewhere in the codebase, so I tried to match it... but it wasn't working.

I'll give 'er another go and come back with more details if I still can't get things lined up correctly.

Reply:
[Should work](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#impl-From%3CError%3E-for-Box%3Cdyn+Error+%2B+Send+%2B+Sync%3E)...

Reply:
I was able to get Into Box working by leaning on `.map_err(|e| anyhow::anyhow!(e.into()))?;`, as while going from Anyhow -> StdErr is easy, the inverse is a bit more verbose...